core/src/main/java/uk/gov/gchq/koryphe/impl/function/ToIterable.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2022 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.koryphe.impl.function;
+
+
+import uk.gov.gchq.koryphe.Since;
+import uk.gov.gchq.koryphe.Summary;
+import uk.gov.gchq.koryphe.function.KorypheFunction;
+
+import java.util.Arrays;
+import java.util.Collections;
+
+@Since("2.4.1")
+@Summary("Converts an object into a Iterable")
+public class ToIterable extends KorypheFunction<Object, Iterable<Object>> {
+    public ToIterable() {
+    }
+
+    public Iterable<Object> apply(final Object value) {
+        final Iterable<Object> rtn;
+        if (null == value) {
+            rtn = Collections.emptyList();
+        } else if (value instanceof Iterable) {
+            //noinspection unchecked
+            rtn = (Iterable<Object>) value;
+        } else if (value.getClass().isArray()) {
+            try {
+                return Arrays.asList((Object[]) value);
+            } catch (final ClassCastException e) {
+                throw new UnsupportedOperationException("The given array is not of type Object[]", e);
+            }
+        } else {
+            rtn = Collections.singletonList(value);
+        }
+        return rtn;
```

Comment:
This isn't consistent with the other Koryphe functions - see [ToList](https://github.com/gchq/koryphe/blob/028e7bf4ecbf48d384bcbae1fcaa9418597e94c6/core/src/main/java/uk/gov/gchq/koryphe/impl/function/ToList.java#L43). 
The test should be `value instanceof Object[]` to see if it's an array of `Object`. There shouldn't be an exception because the cast won't fail. If `value` is an Array of primitives then it should be wrapped in the same way as happens for `ToList`. This is potentially unexpected behaviour, but if that is to be improved/fixed (e.g. so that primitives are supported), it should be done for all of Koryphe and in a separate issue.

Reply:
Is this function not just a duplication of ToList? As it is making new Lists anyway

Reply:
It returns `Iterable<Object>`, whereas ToList returns `List<?>`. I think you might be onto something here though. Instead of creating ToIterable, it might be possible to just cast or otherwise convert the result of ToList into an `Iterable<Object>`. If that can be done easily enough, then this class is redundant and shouldn't be merged.

Reply:
See the comment in [Gaffer PR #2769](https://github.com/gchq/Gaffer/pull/2769/files/bf75e8da67310a514f6408a2c910366d949811a7#r1000270186)